scripts/zones/Crawlers_Nest/npcs/qm8.lua

```diff
@@ -14,46 +14,38 @@ function onTrade(player, npc, trade)
 end
 
 function onTrigger(player, npc)
-
-    local cprog = player:getCharVar("theCrimsonTrial_prog")
-    local cdate = player:getCharVar("theCrimsonTrial_date")
-    local realday = tonumber(os.date("%j")) -- %M for next minute, %j for next day
-
-    if (player:hasKeyItem(tpz.ki.CRAWLER_BLOOD) == true and player:hasKeyItem(tpz.ki.OLD_BOOTS) == true) then
-        player:startEvent(4)
-    elseif (cprog == 1 and cdate == realday) then
-        player:messageSpecial(ID.text.EQUIPMENT_COMPLETELY_PURIFIED)
-    elseif (cprog == 1 and cdate ~= realday) then
-        player:startEvent(5)
+    -- Enveloped in Darkness
+    if player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.ENVELOPED_IN_DARKNESS) == QUEST_ACCEPTED then
+        local timeout = player:getCharVar("envelopedInDarkness_timer")
+        if player:hasKeyItem(tpz.ki.CRAWLER_BLOOD) and player:hasKeyItem(tpz.ki.OLD_BOOTS) then
+            player:startEvent(4) -- Ask to take key items
+        elseif (timeout ~= 0 and timeout > os.time()) then
+            player:messageSpecial(ID.text.EQUIPMENT_NOT_PURIFIED) -- Stagger quest retrieval
+        elseif (timeout ~= 0 and timeout <= os.time()) then
+            player:startEvent(5) -- Finish quest
+        else
+            player:messageSpecial(ID.text.SOMEONE_HAS_BEEN_DIGGING_HERE) -- Player misses key item(s)
+        end
+    -- Default
     else
         player:messageSpecial(ID.text.SOMEONE_HAS_BEEN_DIGGING_HERE)
     end
-
 end
 
 function onEventUpdate(player, csid, option)
 end
 
 function onEventFinish(player, csid, option)
-
     if (csid == 4 and option == 1) then
         player:delKeyItem(tpz.ki.CRAWLER_BLOOD)
         player:delKeyItem(tpz.ki.OLD_BOOTS)
-        player:setCharVar("theCrimsonTrial_date", os.date("%j")) -- %M for next minute, %j for next day
-        player:setCharVar("theCrimsonTrial_prog", 1)
+        player:setCharVar("envelopedInDarkness_timer", os.time() + math.random(10,40)) -- Set timer
         player:messageSpecial(ID.text.YOU_BURY_THE, tpz.ki.OLD_BOOTS, tpz.ki.CRAWLER_BLOOD)
     elseif (csid == 5) then
-        if (player:getFreeSlotsCount() == 0) then
-            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, 14093) -- Warlock's Boots
-        else
-            player:addItem(14093)
-            player:messageSpecial(ID.text.ITEM_OBTAINED, 14093) -- Warlock's Boots
-            player:setCharVar("theCrimsonTrial_date", 0)
-            player:setCharVar("theCrimsonTrial_prog", 0)
-            player:setCharVar("needs_crawler_blood", 2) -- Fixed being unable start next quest
-            player:addFame(SANDORIA, 40)
-            player:completeQuest(SANDORIA, tpz.quest.id.sandoria.ENVELOPED_IN_DARKNESS)
-        end
+        npcUtil.completeQuest(player, SANDORIA, tpz.quest.id.sandoria.ENVELOPED_IN_DARKNESS, {
```

Comment:
does this work without having `require("scripts/globals/npc_util")` header?

Reply:
Funny you mention it, because I tested this extensively. Yes it seems so. Should I rather include it anyways?

Reply:
@TeoTwawki  what u think?

Reply:
Btw the parentheses were removed because I shortened the superfluity of `:hasKeyItem() == true`

Reply:
no need for all if ( ) parenthesis even if it has an `and` or `or` unless it is really needed for order of resolution

Reply:
Sorry, what do you refer to? The remaining parenthesis of the `elseif` blocks inside `onTrigger()`? I gladly take them out actually...

Reply:
Since I couldn't find any file where npc_util is not explicitly required I also added it accordingly. Marking all of this as resolved then.

Reply:
It wasn't a prob in testing was because something else happened to be loaded that already required it, putting it in memory for this script by chance. Always require teh things you use :+1:  (and only the ones you use - have seen ppl just add the kitchen sink before lol)

as for parentheses what kain means is the outmost ones are usually not needed:
```lua
if condition1 and condition2 then
```
vs:
```lua
if condition1 and condition2 then
```
If things get complicated we'll wrap things in parenthese but otherwise they only are involved in function calls: `entity:doStuff(arguments)`

Reply:
repeat because for the life of me I cannot type correctly the example in one try >.< so sorry


as for parentheses what kain means is the outmost ones are usually not needed:
```lua
if (condition1 and condition2) then
```
vs:
```lua
if condition1 and condition2 then
```
If things get complicated we'll wrap things in parentheses but otherwise they only are involved in function calls: `entity:doStuff(arguments)`